aspnetcore/includes/blazor-prerendering.md

```diff
@@ -11,17 +11,25 @@ To delay JavaScript interop calls until after the connection with the browser is
 @code {
     private ElementReference myInput;
 
-    protected override void OnAfterRender(bool firstRender)
+    protected override async Task OnAfterRenderAsync(bool firstRender)
     {
         if (firstRender)
         {
-            JSRuntime.InvokeVoidAsync(
+            await JSRuntime.InvokeVoidAsync(
                 "setElementValue", myInput, "Value set after render");
         }
     }
 }
 ```
 
+For the preceding example code, provide a `setElementValue` JavaScript function inside the `<head>` element of *wwwroot/index.html* (Blazor WebAssembly) or *Pages/_Host.cshtml* (Blazor Server). The function is called with `IJSRuntime.InvokeVoidAsync` and doesn't return a value:
+
+```html
+<script>
+  window.setElementValue = (elem, val) => elem.value = val;
```

Comment:
I'm worried that this is showing an anti-pattern. By manipulating the DOM directly in a way that Blazor is unaware of, you run the risk of messing up Blazor's diffing algorithm. I better example might be initializing some sort of JS component, but I don't know what a simple example might look like. @pranavkm @javiercn @SteveSandersonMS any ideas?

Reply:
We can always add a note that this is for demo purposes only. We've done that a few times with good outcomes AFAIK. I recommend an explicit remark in the topic and a code comment to that effect.

Reply:
Yeah, I would suggest two things: 1. change the DOM element from an input to a static div (so folks don't try to bind to it or do other Blazor-y things), and 2. Add a note warning folks that modifying the DOM directly with JS can interfere with Blazor's change tracking.